server/src/main/java/org/opensearch/wlm/cancellation/LongestRunningTaskFirstSelectionStrategy.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.cancellation;
+
+import org.opensearch.tasks.Task;
+
+import java.util.Comparator;
+
+/**
+ * Represents a task selection strategy that prioritizes the longest running tasks first.
+ */
+public class LongestRunningTaskFirstSelectionStrategy extends AbstractTaskSelectionStrategy {
+
+    /**
+     * Returns a comparator that sorts tasks based on their start time in descending order.
+     *
+     * @return The comparator
+     */
+    @Override
+    public Comparator<Task> sortingCondition() {
+        return Comparator.comparingLong(Task::getStartTime);
+    }
+}
```

Comment:
Used class? Can we remove this?

Reply:
Agree, we should only keep one implementation in codebase. Can we provide a brief class level comment on why a certain strategy is chosen ?

Reply:
discussing the same in this comment - https://github.com/opensearch-project/OpenSearch/pull/15151#discussion_r1717517819
so resolving this for now.